test/parallel/test-crypto-argon2.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const { hasOpenSSL } = require('../common/crypto');
+
+if (!hasOpenSSL(3, 2))
+  common.skip('requires OpenSSL >= 3.2');
+
+const assert = require('node:assert');
+const crypto = require('node:crypto');
+
+function runArgon2(options) {
+  const syncResult =
+    crypto.argon2Sync(options);
+
+  crypto.argon2(options,
+                common.mustSucceed((asyncResult) => {
+                  assert.deepStrictEqual(asyncResult, syncResult);
+                }));
+
+  return syncResult;
+}
+
+const message = Buffer.alloc(32, 0x01);
+const nonce = Buffer.alloc(16, 0x02);
+const secret = Buffer.alloc(8, 0x03);
+const associatedData = Buffer.alloc(12, 0x04);
+const defaults = { message, nonce, parallelism: 4, tagLength: 32, memory: 64 << 10, passes: 3, type: 'argon2id' };
+
+const good = [
+  [
+    { type: 'argon2d' },
+    'bf37a2a7530e053a8bd2784a9d50dc7de451e33bd581096922bc7f9ef66020ed',
+  ],
+  [
+    { type: 'argon2i' },
+    '96334882febdd85eb9b2cf367479fbad2d5b87cf79f9076f51b23589560a2e0a',
+  ],
+  [
+    { type: 'argon2id' },
+    'fedd802b86b17230843c6d3f025c81d3f472fbf9daaf26897fa88844732167ec',
+  ],
+  [
+    { secret },
+    '71d8a1361e3660f8e8158021a6870f223d42342c6a194429252c6d6ad19c25d6',
+  ],
+  [
+    { associatedData },
+    '7da97d5dd2ba433dbb1b26e00a496caa673501b9a437e863855201eb51cea6ef',
+  ],
+  [
+    { secret, associatedData },
+    '47a677ebc68412b8e17949c921fdca065352114ffb2a09eebe61adc451eecf1e',
+  ],
+  [
+    { passes: 2 },
+    '2139b90863e8bbffff1459c9f08f6460db08def592991e78f11b6c5765bd1fac',
+  ],
+  [
+    { memory: 32 << 10 },
+    '2beec876bf5b1098e8349d094385eda1dd472733d103e0f1afad4d28f948e808',
+  ],
+];
```

Comment:
I think I'll replace these with [test vectors from OpenSSL](https://github.com/openssl/openssl/blob/openssl-3.2/test/recipes/30-test_evp_data/evpkdf_argon2.txt) for a "nothing up my sleeve" approach.